extensions/vertx/runtime/src/main/java/io/quarkus/vertx/runtime/jackson/QuarkusJacksonJsonCodec.java

```diff
@@ -30,11 +30,31 @@
  */
 class QuarkusJacksonJsonCodec implements JsonCodec {
 
-    private static final ObjectMapper mapper;
+    private static volatile ObjectMapper mapper;
     // we don't want to create this unless it's absolutely necessary (and it rarely is)
     private static volatile ObjectMapper prettyMapper;
 
     static {
+        populateMapper();
+    }
+
+    public static void reset() {
+        mapper = null;
+        prettyMapper = null;
+    }
+
+    private static ObjectMapper mapper() {
+        if (mapper == null) {
+            synchronized (QuarkusJacksonJsonCodec.class) {
+                if (mapper == null) {
+                    populateMapper();
+                }
+            }
+        }
+        return mapper;
+    }
+
+    private static void populateMapper() {
```

Comment:
Will the `QuarkusJacksonJsonCodec` always be loaded by Vert.x? Because if so, I wonder if we should populate a new mapper directly instead of resetting it.
That being said I'm not entirely sure we could safely drop some of the guards but wanted to push the idea in case you didn't think about it.

Reply:
The name of the method is probably confusing here - what it does is actually create a new `ObjectMapper` used by Vert.x. Do you want me to change the method as to make it easier for us to understand next time we stumble upon the code?

Reply:
No I understand what the method does.

What I was wondering is if the `reset()` could `populate()` the default mapper instead of populating it. Maybe in a synchronized method so that we could drop the volatile.

That would make sense only if the codec is always initialized though. Otherwise we might pay the initialization cost for nothing.

It might be a bad idea though, I will let you decide what's best.

Reply:
> could populate() the default mapper instead of populating it. Maybe in a synchronized method so that we could drop the volatile.

I want to avoid that because populating means extracting from Arc, which I am not 100% sure would give the proper bean at this point.
Whereas the way it's done now, the loading of the mapper happens in the same "timing" as if there no reload.

Reply:
OK, sure, it makes perfect sense. We can revisit if we see it creates some performance issues but in this case, we will have to be very careful as to how operations are ordered.